packages/squiggle-lang/src/rescript/Utility/E/E_A.res

```diff
@@ -305,56 +305,50 @@ module Floats = {
     /*
       This function goes through a sorted array and divides it into two different clusters:
       continuous samples and discrete samples. The discrete samples are stored in a mutable map.
-      Samples are thought to be discrete if they have any duplicates.
+      Samples are thought to be discrete if they have at least `minDiscreteWight` duplicates.
+
+      If the min discreet weight is 4, that would mean that at least four elements needed from a specific
+      value for that to be kept as discrete. This is important because in some cases, we can expect that
+      some common elements will be generated by regular operations. The final continous array will be sorted.
  */
-    let _splitContinuousAndDiscreteForDuplicates = (sortedArray: array<float>) => {
+    let splitContinuousAndDiscreteForMinWeight = (
+      sortedArray: array<float>,
+      ~minDiscreteWeight: int,
+    ) => {
       let continuous: array<float> = []
       let discrete = FloatFloatMap.empty()
-      Belt.Array.forEachWithIndex(sortedArray, (index, element) => {
-        let maxIndex = (sortedArray |> Array.length) - 1
-        let possiblySimilarElements = switch index {
-        | 0 => [index + 1]
-        | n if n == maxIndex => [index - 1]
-        | _ => [index - 1, index + 1]
-        } |> Belt.Array.map(_, r => sortedArray[r])
-        let hasSimilarElement = Belt.Array.some(possiblySimilarElements, r => r == element)
-        hasSimilarElement
-          ? FloatFloatMap.increment(element, discrete)
-          : {
-              let _ = Js.Array.push(element, continuous)
+
+      let flush = (cnt: int, value: float): unit => {
+        if cnt >= minDiscreteWeight {
+          FloatFloatMap.add(value, cnt->Belt.Int.toFloat, discrete)
+        } else {
+          for _ in 1 to cnt {
+            let _ = continuous->Js.Array2.push(value)
+          }
+        }
+      }
+
+      if sortedArray->Js.Array2.length != 0 {
```

Comment:
For things like this, I find it more readable without the arrows.

```
is Js.Array2.length(sortedArray) != 0 {
```

Also, would be good to map to E.A

Reply:
It could also be nice to have an E.isEmpty method

Reply:
I've been avoiding `E.A.*` until #1134 is fixed, though `length` should be safe enough.

Actually, II think this `if` is not necessary, I'll just delete it.

Noted about `Js.Array2.length`; my intuition was the opposite, I'm usually using `->` whenever the function seems like a method (and it's a method/object property in JS).

Reply:
> I've been avoiding E.A.*
I'd recommend updating this gradually, instead of holding out at all for some big refactor. 

>  I'm usually using -> whenever the function seems like a method. 
I usually do as well, but find that comparisons are one exception. It's a bit of a style guide issue though.